### PR TITLE
Fetcher

### PR DESCRIPTION
**Pull request by alexander871015**
Finished writing dependency rules for bazel for all the necessary external libraries

### CODE REVIEW

File: hologram_dashboard/fetcher/enums.proto [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+*/
+
+syntax = "proto2";
+package hologramAvailability;
+
+enum DataType {
+    PLAY_COUNTRY = 0;
```
Comment (sarahchen-google) [inline hologram_dashboard/fetcher/enums.proto:21]: In proto enum, 0 is used for UNKNOWN_DATA_TYPE/UNSPECIFED_DATA_TYPE/DEFAULT_DATA_TYPE.

Valid enum values should be >0.

File: hologram_dashboard/fetcher/enums.proto [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+ }
+
+ enum StatusType {
+     UNKNOWN = 0;
+     SUCCESS = 1;
+     DATA_NOT_INJECTED = 2;
+ }
+
+ enum Pipeline {
+    HOLOGRAM_DEFAULT = 0;
```
Comment (sarahchen-google) [inline hologram_dashboard/fetcher/enums.proto:46]: Same above.

File: hologram_dashboard/fetcher/fetcher.cc [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+*/
+#include "fetcher.h"
+
+int main() {
+    std::cout << "Hi\n";
```
Comment (sarahchen-google) [inline hologram_dashboard/fetcher/fetcher.cc:19]: We need to add a new file, like fetcher_main.cc, to put the main function.

File: hologram_dashboard/fetcher/fetcher.h [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    limitations under the License.
+*/
+#include <iostream>
+#include <string>
+#include <unordered_map>
+#include <gflags/gflags.h>
+#include "fetcher/hologram_availability.pb.h"
+#include "fetcher/hologram_config.pb.h"
+
+using namespace std;
```
Comment (sarahchen-google) [inline hologram_dashboard/fetcher/fetcher.h:23]: Remove this line and use std:: below?

File: hologram_dashboard/fetcher/fetcher.h [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+class HologramAvailabilityFetcher {
+    public:
+        HologramAvailabilityFetcher();
+    private:
+        unordered_map<string, string> kvick_config;
+		unordered_map<string, string> kvick_time;
+		unordered_map<string, string> kvick_location;
```
Comment (sarahchen-google) [inline hologram_dashboard/fetcher/fetcher.h:30]: Fix indent please.

File: hologram_dashboard/fetcher/hologram_availability.proto [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+*/
+
+syntax = "proto2";
+package hologramAvailability;
```
Comment (sarahchen-google) [inline hologram_dashboard/fetcher/hologram_availability.proto:18]: Use 

package wireless_android_play_analytics;

File: hologram_dashboard/fetcher/hologram_config.proto [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+*/
+
+syntax = "proto2";
+package hologramAvailability;
```
Comment (sarahchen-google) [inline hologram_dashboard/fetcher/hologram_config.proto:18]: Same above.

File: hologram_dashboard/fetcher/fetcher.cc [hunk 8]
```diff
@@ -0,0 +1,5 @@
+#include "fetcher_lib/hologram_availability_fetcher.h"
```
Comment (sarahchen-google) [inline hologram_dashboard/fetcher/fetcher.cc:1]: Name it fetcher_main.cc to make it more clear to readers.

File: hologram_dashboard/fetcher/fetcher_lib/enums.proto [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+    DEVICE_INTEGRITY_SOURCE = 7; 
+    SHIELD_ID = 8;
+    SERVER_EXPERIMENT = 9; 
+    APPS_TITLE_TIER = 10;
+    APP_TYPE_SOURCE = 11;
+    ANDROID_SPAM_ABUSE = 12;
+    USER_EXPERIMENT_ID_AND_TEST_CODE = 13;
+    USER_ATTRIBUTE = 14;
+    PRIME_MERIDIAN= 15;
+    PLAY_COUNTRY = 16;
```
Comment (sarahchen-google) [inline hologram_dashboard/fetcher/fetcher_lib/enums.proto:37]: Can you sort them by alphabetical order?

File: hologram_dashboard/fetcher/fetcher_lib/hologram_availability_fetcher.h [hunk 10]
```diff
@@ -0,0 +1,149 @@
...
+#pragma once   
+#include <iostream>
+#include <string>
+#include <unordered_map>
+#include <gflags/gflags.h>
+#include "fetcher/fetcher_lib/hologram_availability.pb.h"
+#include "fetcher/fetcher_lib/hologram_config.pb.h"
+
+
+DEFINE_string(chipper_server, "undefined", "Specifies the server on which chipper file location is stored");
```
Comment (sarahchen-google) [inline hologram_dashboard/fetcher/fetcher_lib/hologram_availability_fetcher.h:25]: -> chipper_batch_job_cell

File: hologram_dashboard/fetcher/fetcher_lib/hologram_availability_fetcher.h [hunk 11]
```diff
@@ -0,0 +1,149 @@
...
+DEFINE_string(chipperGDPR_server, "undefined", "Specifies the server on which chipperGDPR file location is stored");
```
Comment (sarahchen-google) [inline hologram_dashboard/fetcher/fetcher_lib/hologram_availability_fetcher.h:26]: diitto

File: hologram_dashboard/fetcher/fetcher_lib/hologram_availability_fetcher.h [hunk 12]
```diff
@@ -0,0 +1,149 @@
...
+DEFINE_string(prime_meridian_server, "undefined", "Specifies the server on which prime meridian file location is stored");
```
Comment (sarahchen-google) [inline hologram_dashboard/fetcher/fetcher_lib/hologram_availability_fetcher.h:27]: doitto

File: hologram_dashboard/fetcher/fetcher_lib/hologram_availability_fetcher.h [hunk 13]
```diff
@@ -0,0 +1,149 @@
+/*
```
Comment (sarahchen-google) [inline hologram_dashboard/fetcher/fetcher_lib/hologram_availability_fetcher.h:1]: I don't quite get what you want to here for fetcher. Can you grab some time to sync with me?

File: hologram_dashboard/fetcher/fetcher.cc (hunk 8 above)
Comment (alexander871015) [reply hologram_dashboard/fetcher/fetcher.cc:1]: This is the library, the main would reside in fetcher.cc should i change fetcher.cc to fetcher_main instead?

**Pull request closed without merging**